rust/src/lib/nm/ip.rs

```diff
@@ -62,7 +62,7 @@ fn gen_nm_ipv4_setting(
         // enabled.
         nm_setting.routes = Vec::new();
     }
-    if iface_ip.enabled && !iface_ip.dhcp {
+    if !iface_ip.enabled || !iface_ip.dhcp {
```

Comment:
This is hard to read for mortal like me.

Please use this one and keep the old code.
```rust
if !iface_ip.enabled {
    nm_setting.routes = Vec::new();
}
```

Reply:
It's even hard for me after the weekend :-) I will change it.